File: tests/javascript/third_parties/algolia/testdata/datatype_in_save_object.js
const algoliasearch = require("algoliasearch")
const myAlgolia = algoliasearch("123", "123")

const index = myAlgolia.initIndex("test_index")

// saveObject
const userObj = { user_id: user.ip_address }
// bearer:expected javascript_third_parties_algolia
index
  .saveObject(userObj, { autoGenerateObjectIDIfNotExist: true })
  .then(console.log("obj saved"))

// bearer:expected javascript_third_parties_algolia
index.saveObjects([{ email: user.email }])

File: tests/ruby/third_parties/algolia/testdata/datatype_in_save_object.rb
client = Algolia::Search::Client.create('YourApplicationID', 'YourWriteAPIKey')
index = client.init_index("my_index")

# bearer:expected ruby_third_parties_algolia
index.save_object({ email: user.email }, { auto_generate_object_id_if_not_exist: true })

# bearer:expected ruby_third_parties_algolia
index.save_objects([{ email: user.email }], { auto_generate_object_id_if_not_exist: true })
